ocaml/networkd/bin/network_monitor_thread.ml

```diff
@@ -371,9 +371,22 @@ let clear_input fd =
   in
   Unix.set_nonblock fd ; loop () ; Unix.clear_nonblock fd
 
+(* Watch for changes in the network interfaces: whether the IP changes, and
+   whether the links go online or offline *)
+let filters =
```

Comment:
Does this filter out events related to interfaces the toolstack doesn't need to monitor? Otherwise it's going to signal a networking change every time something happens to the vif and tap interfaces.

Reply:
will test that, sounds like xapi is signaled way too many times

Reply:
I don't think that vif/tap devices would turn up in the output from {{ip monitor address}}, because they don't have IP addresses in dom0.

Reply:
Code now ignore vifs and tap devices.

I think the output processing of ip monitor can be fragile, monitoring the sysfs will can be much more robust. I wouldn't try to use angstrom to try to parse the output, seems overkill when monitoring seems easier

Reply:
Or we could switch to NETLINK directly rather than `ip`.

Reply:
Ah, didn't know we had these already: https://github.com/xapi-project/ocaml-netlink

Reply:
Yes, it was my first attempt at using ctypes :) It's not complete bindings though. The bit of Netlink that we need isn't there now. But I always wanted to extend it and make more use of it in xcp-network, to replace `ip` calls.